### PR TITLE
refactor(packages/linker) Skip same origin

### PR DESCRIPTION
**Pull request by 9renpoto**
# TL;DR

- Added test
- Skip same origins

### CODE REVIEW

File: packages/linker/test/handler.test.ts [hunk 1]
```diff
@@ -1,97 +1,140 @@
...
   })
 
   it('not cross domain', () => {
-    const url = document.location.href
-    const handler = submit([domain], agent.get('linkerParam'))
-    const form = createForm(document.location.href, 'post')
-    handler({ target: form } as any)
-    assert(url === form.action, 'not added query string')
+    const l = toLink(internet.url())
+    fixture.set(`<form method="get" action="${l.href}"><form>`, true)
```
Comment (shoota) [inline packages/linker/test/handler.test.ts]: query params will not add if submit `method='get'` form.

Comment (9renpoto) [reply packages/linker/test/handler.test.ts]: see https://github.com/userdive/agent.js/pull/368/files#diff-fa2b92802143c2dc45119226f16189cbR120

File: packages/linker/src/provide.ts [hunk 2]
```diff
@@ -1,10 +1,10 @@
-import { q } from 'userdive'
+import { namespace, q } from 'userdive'
```
Comment (shoota) [inline packages/linker/src/provide.ts]: 🎉

File: packages/linker/src/handler.ts [hunk 3]
```diff
@@ -47,13 +50,14 @@ function addableForm (domains: Domain[], element: any) {
   return match ? matchDomain(domains, match[1]) : false
 }
 
-function matchDomain (domains: Domain[], test: string): boolean {
-  if (test === document.location.hostname) {
+const re = new RegExp(location.hostname)
+function matchDomain (domains: Domain[], href: string): boolean {
```
Comment (shoota) [inline packages/linker/src/handler.ts]: `matchDomain` is using to check href and _action_.

File: packages/linker/test/handler.test.ts (hunk 1 above)
Comment (shoota) [reply packages/linker/test/handler.test.ts]: Yes, it is good and working. But, i think this is not `not cross domain` test case.

**Pull request merged**